libs/micronaut-console/src/main/groovy/com/agorapulse/micronaut/console/http/ConsoleHeadersFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2020-2021 Agorapulse.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.agorapulse.micronaut.console.http;
+
+import com.agorapulse.micronaut.console.ConsoleConfiguration;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.FilterChain;
+import io.micronaut.http.filter.HttpFilter;
+import io.reactivex.Flowable;
+import org.reactivestreams.Publisher;
+
+@Filter("${console.path:/console}/**")
+@Requires(property = "console.header-name")
+public class ConsoleHeadersFilter implements HttpFilter {
+
+    private final ConsoleConfiguration configuration;
+
+    public ConsoleHeadersFilter(ConsoleConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Publisher<? extends HttpResponse<?>> doFilter(HttpRequest<?> request, FilterChain chain) {
+        return Flowable.just(request)
+            .switchMap(req -> {
+                if (HttpMethod.POST.equals(req.getMethod())) {
+                    String headerValue = req.getHeaders().get(configuration.getHeaderName());
+                    if (headerValue == null) {
+                        return Flowable.just(HttpResponse.status(HttpStatus.FORBIDDEN, "Missing verification header"));
+                    }
+
+                    if (configuration.getHeaderValue() != null &&!configuration.getHeaderValue().equals(headerValue)) {
```

Comment:
As I understand If there is a `headerName` and no `headerValue` in the configuration, the validation will be bypassed.  Is that intended ?

Reply:
in that case the value of the header does not matter but the header must be present

Reply:
Should we force to have a `headerValue` in the configuration, just to be safe ?

Reply:
I left it intentionally nullable but If you think it's a bad idea I can force it.

Reply:
👍

Reply:
Sorry, my 👍  wasn't clear. 
I think it would be better to force it. I am afraid that the value will be often "forget".

Reply:
yes, it wasn't 😄 please, see the recent changes in 859421c